rpc/wrtc_server_stream.go

```diff
@@ -292,6 +292,15 @@ func (s *webrtcServerStream) processHeaders(headers *webrtcpb.RequestHeaders) {
 		return
 	}
 
+	md, ok := metadata.FromIncomingContext(s.ctx)
+	if !ok {
+		md = metadata.New(map[string]string{})
+	}
+	md.Append("trace-id", headers.Metadata.Md["trace-id"].Values[0])
```

Comment:
It looks this line is causing the test failure in our [GitHub actions](https://github.com/viamrobotics/goutils/actions/runs/5404691236/jobs/9819197882?pr=192#step:8:1158)
```
panic: runtime error: invalid memory address or nil pointer dereference
```
It may be because the `Md` isn't guaranteed to have any of these values.